lib/link_monitor/upsert_resource_monitor.rb

```diff
@@ -0,0 +1,42 @@
+module LinkMonitor
+  class UpsertResourceMonitor
+    def initialize(links:, service:, resource_type:, resource_id:)
+      @links = links
+      @service = service
+      @resource_type = resource_type
+      @resource_id = resource_id
+    end
+
+    def call
+      monitor = ResourceMonitor.find_or_create_by(service: service, resource_type: resource_type, resource_id: resource_id)
```

Comment:
We do a find by 3 different properties here, do we expect the service or resource type to change but still want to use the old ResourceMonitor record?

With this current implementation, will there be instances where we create a new ResourceMonitor and do not clean up and old one?

Reply:
It seems to me that applications shouldn't affect the `ResourceMonitor`s of others, so `service` makes sense to me here. But I don't quite see the advantage of having both a `resource_type` and a `resource_id`. It feels like we can have a `service` and a `name`.

Reply:
ResourceMonitor will be per edition on whitehall, so need to uniquely identify them which is why resource_id probably exists.

If `application` is used over `service` in the GDS world we should probably be consistent with that.

Reply:
Hmm, this feels to me like Link Checker API knowing too much about the caller of the API. I imagine this could work by just having Whitehall provide a `name` (or `resource_type`) of `edition:4322` - and similar for Local Links Manager.

Reply:
Yeah I'd tend to agree, maybe `reference` would be better than `name`, but yeah, leave it up to the caller to ensure it's unique.

Reply:
Yeah `reference` works well, it's what GOV.UK Notify uses in fact.